csrc/scheduler/utils.cpp

```diff
@@ -2468,15 +2467,13 @@ void prepareForMemoryTypePromotion(Fusion* fusion) {
         producer->getMemoryType());
 
     // If already placed on Global, nothing to worry about
-    if (producer->getMemoryType() == MemoryType::Global ||
-        cached.count(producer) != 0) {
+    if (producer->getMemoryType() == MemoryType::Global) {
```

Comment:
So you are creating multiple global memory caches for multiple slice? This does not sound optimal. Slice of the same producer should use the same global memory cache. I think you can do so by maintaining a `producer-> copy_of_producer` mapping.

Reply:
Ah, that's a very good point. Will update the code.

Reply:
Updated